### ISSUE TITLE
CurvedShapes corrupts files in console mode: <class 'AttributeError'>: module 'FreeCADGui' has no attribute 'addCommand'

### ISSUE THREAD

**Issue by AIRCAP** [chbergmann/CurvedShapesWorkbench#33]
Freecad has a batch/console mode. This is useful for example to automatically execute a standalone script such as
1. load a freecad file with a parametrized part
2. load parameters from a CSV file - or piped from another program
3. export the parametrized shape as a mesh
4. end

Usage: freecadcmd <Filename.FCStd> <Macro.FCMacro>
or
freecad -c <Filename.FCStd> <Macro.FCMacro>
In my case I use this workflow to automatically adapt a wing profile based on Cfd results in open foam

Issue: It does not work if the freecad file uses any parts from CurvedShapes. In console mode, GUI elements are not available, freecad extensions and macros typically put calls to GUI methods in try: catch blocks, but CurvedShapes assumes the GUI is always available.

when loading test.FCStd from reproduce.zip below, it loads fine:
```
freecad test.FCStd
FreeCAD 0.21.2, Libs: 0.21.2R33771 (Git)
© Juergen Riegel, Werner Mayer, Yorik van Havre and others 2001-2023
FreeCAD is free and open-source software licensed under the terms of LGPL2+ license.
FreeCAD wouldn't be possible without FreeCAD community.
  #####                 ####  ###   ####  
  #                    #      # #   #   # 
  #     ##  #### ####  #     #   #  #   # 
  ####  # # #  # #  #  #     #####  #   # 
  #     #   #### ####  #    #     # #   # 
  #     #   #    #     #    #     # #   #  ##  ##  ##
  #     #   #### ####   ### #     # ####   ##  ##  ##

```
![test](https://github.com/user-attachments/assets/34d1bb62-d81f-4193-87cb-0e2754f9f724)

However, when loading in console mode, this happens:

```
freecad test.FCStd -c
Traceback (most recent call last):
  File "/home/eprice/.local/share/FreeCAD/Mod/CurvedShapes/./CurvedSegment.py", line 473, in <module>
    FreeCADGui.addCommand('CurvedSegment', CurvedSegment())
<class 'AttributeError'>: module 'FreeCADGui' has no attribute 'addCommand'
Importing project files......
Postprocessing......				(100 %)		
[FreeCAD Console mode <Use Ctrl-D (i.e. EOF) to exit.>]
```
making modifications to the file (such as parameter changes) and saving in console mode then breaks any curved surface parts, they become regular compounds of their respective curves, but no boundary shapes

thisbreaks.FCMacro:
```
import FreeCAD

FreeCAD.ActiveDocument.save()
```

```
cp test.FCStd broken.FCStd
freecadcmd broken.FCStd thisbreaks.FCMacro
FreeCAD 0.21.2, Libs: 0.21.2R33771 (Git)
(c) Juergen Riegel, Werner Mayer, Yorik van Havre and others 2001-2023
FreeCAD is free and open-source software licensed under the terms of LGPL2+ license.
FreeCAD wouldn't be possible without FreeCAD community.
  #####                 ####  ###   ####  
  #                    #      # #   #   # 
  #     ##  #### ####  #     #   #  #   # 
  ####  # # #  # #  #  #     #####  #   # 
  #     #   #### ####  #    #     # #   # 
  #     #   #    #     #    #     # #   #  ##  ##  ##
  #     #   #### ####   ### #     # ####   ##  ##  ##

Traceback (most recent call last):
  File "/home/eprice/.local/share/FreeCAD/Mod/CurvedShapes/./CurvedSegment.py", line 473, in <module>
    FreeCADGui.addCommand('CurvedSegment', CurvedSegment())
<class 'AttributeError'>: module 'FreeCADGui' has no attribute 'addCommand'
Importing project files......
Postprocessing......				(100 %)
```

see attached files for reproduction

[reproduce.zip](https://github.com/user-attachments/files/17010710/reproduce.zip)

**Comment by AIRCAP** [comment]
Add. try: catch is probably not the right way. I checked some other extensions, they put all calls to FreeCAD gui in GuiUp checks:


like this:
```
if FreeCAD.GuiUp:
    import FreeCADGui
```

**Comment by AIRCAP** [comment]
I managed to get rid of the errormessage by adding GuiUp checks to the files within the module, but the freecad file still breaks when saved in console mode, since now CurvedShapes simply doesnt get initilalized. so sadly not such a simple fix

**Comment by AIRCAP** [comment]
comparing https://wiki.freecad.org/Scripted_objects with the code in CurvedShapes - taking CurvedSegment as example, the "object" defined is called `"Part::FeaturePython","CurvedSegment" ` this implies that the `class CurvedSegment` is the object class, - while the GUI command would normally be called `class CurvedSegmentCommand` -- instead class CurvedSegment is the gui command and the object class is `class CurvedSegmentWorker` -- for some reason when loaded with GUI FreeCAD has no issue with that, but when the Gui is not loaded, it forgets what class this thing is... very very weird...  I don#t think renaming the class alone fixes things - that might actually break backwards-compatibility - it looks like the serialization includes the reference to CurvedSegmentWorker...  I'm confused.

**Comment by AIRCAP** [comment]
OK, I looked itno this some more. Turns out what happens if you save a FreeCAD file in console mode, the viewproviders are lost. All of them - not just CurvedArray but for everything. Then on the next load all objects become generic objets and the freecad UI becomes useless.  This is obviously a bug/shortcoming of FreeCAD.  Nevertheless there is still a bug in CurvedShapes causing it to not load correctly when the UI is not loaded. I'm preparing a pull request
